### PR TITLE
docs(example): adds virtual fields example

### PR DESCRIPTION
**Pull request by JessRynkar**
## Description

Adds virtual fields example.

- [X] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [X] Documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Existing test suite passes locally with my changes
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

File: examples/virtualFields/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Copy this example with [Degit](https://www.npmjs.com/package/degit) by running the following command at your terminal:
+
+```bash
+npx degit github:payloadcms/payload/examples/virtualFields
+```
+
+### Step 2. Set up the environment variables
+
+Copy the `.env.example` file in this directory to `.env`:
```
Comment (JarrodMFlesch) [inline examples/virtualFields/README.md:30]: - [ ] `.env.example` is missing in this folder

File: examples/virtualFields/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+Seeding will drop any existing database and add example data. You only need to do this once on startup, or if you want to reset your data at any point.
+
+### Step 4. Start development server
+
+Start the development server with:
+```bash
+yarn dev
+```
+
```
Comment (JarrodMFlesch) [inline examples/virtualFields/README.md:54]: - [ ] If they used the seed script, what are the credentials? Lets list them here to reduce friction.

- [ ] Lets add the BeforeLogin component used in the [demo repo](https://github.com/payloadcms/public-demo/tree/master/src/components/BeforeLogin). You can conditionally render inside the component if `process.env.PAYLOAD_PUBLIC_SEED === "true"`

File: examples/virtualFields/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "virtual-fields",
+  "version": "1.0.0",
+  "main": "dist/server.js",
+  "license": "MIT",
+  "scripts": {
+    "seed": "cross-env PAYLOAD_SEED=true PAYLOAD_CONFIG_PATH=src/payload.config.ts nodemon",
```
Comment (JarrodMFlesch) [inline examples/virtualFields/package.json:7]: - [ ] Rename `PAYLOAD_SEED` to `PAYLOAD_PUBLIC_SEED`

File: examples/virtualFields/src/server.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  await payload.init({
+    secret: process.env.PAYLOAD_SECRET,
+    mongoURL: process.env.MONGODB_URI,
+    express: app,
+    onInit: async () => {
+      payload.logger.info(`Payload Admin URL: ${payload.getAdminURL()}`);
+    },
+  });
+
+  if (process.env.PAYLOAD_SEED === 'true') {
```
Comment (JarrodMFlesch) [inline examples/virtualFields/src/server.ts:28]: - [ ] Rename `PAYLOAD_SEED` to `PAYLOAD_PUBLIC_SEED`

**Review by JarrodMFlesch — CHANGES_REQUESTED** [review]
These comments are geared toward documentation and reducing friction when booting up the demo.

File: examples/virtualFields/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+## Running locally
+
+[- Step 1. Copy example](#step-1-copy-example)
+
+[- Step 2. Set up the environment variables](#step-2-set-up-the-environment-variables)
+
+[- Step 3. Add dependencies and seed data](#step-3-add-dependencies-and-seed-data)
+
+[- Step 4. Start development server](#step-4-start-development-server)
+
```
[the remark this answers is not in the document]
Comment (JessChowdhury) [reply examples/virtualFields/README.md:19]: I added some more on the collections to the next section 'Walkthrough' - might still need more detail let me know what you think.

**Comment by jacobsfletch** [comment]
Should the directory structure be `examples/virtual-fields`? Instead of camel-cased.

File: examples/virtual-fields/README.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+- Yarn or NPM
+- NodeJS version 10+
+- A Mongo Database - IMPORTANT: you need to either have MongoDB running locally, or have signed up for a free MongoDB Atlas server in order to run this example locally.
+
+## Getting Started
+
+First copy this example with [Degit](https://www.npmjs.com/package/degit) by running the following command at your terminal:
+
+```bash
+npx degit github:payloadcms/payload/examples/virtualFields
```
Comment (jacobsfletch) [inline examples/virtual-fields/README.md:15]: This path is now wrong.

Comment (JessChowdhury) [reply examples/virtual-fields/README.md]: Fixed

File: examples/virtual-fields/README.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
+# Virtual Fields Example for Payload CMS
+
+This example demonstrates multiple use-cases for virtual fields.
+
+The following software is required:
```
[the remark this answers is not in the document]
Comment (JessChowdhury) [reply examples/virtual-fields/README.md]: Removed

File: examples/virtual-fields/src/seed/index.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import payload from 'payload';
+import { MongoClient } from 'mongodb';
+import { eventsOne, eventsTwo } from './events';
+import { locationOne, locationTwo } from './locations';
+import { staffOne, staffTwo } from './staff';
+
+export async function seed() {
+  try {
+    payload.logger.info('Dropping database...');
```
Comment (jacobsfletch) [inline examples/virtual-fields/src/seed/index.ts:9]: No need to drop the db here, just add `PAYLOAD_DROP_DATABASE=true` to your dev script.

File: examples/virtual-fields/src/server.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    secret: process.env.PAYLOAD_SECRET,
+    mongoURL: process.env.MONGODB_URI,
+    express: app,
+    onInit: async () => {
+      payload.logger.info(`Payload Admin URL: ${payload.getAdminURL()}`);
+    },
+  });
+
+  if (process.env.PAYLOAD_PUBLIC_SEED === 'true') {
+    await seed();
```
Comment (jacobsfletch) [inline examples/virtual-fields/src/server.ts:29]: Add `payload.logger.info('---- SEEDING DATABASE ----')` just before this line.

**Review by jacobsfletch — APPROVED** [review]

**Pull request merged**